internal/test/wmcb/wmcb_test.go

```diff
@@ -635,30 +681,47 @@ func testWMCBCluster(t *testing.T) {
 	assert.Lenf(t, winNodes.Items, 1, "expected one node to have node label but found: %v", len(winNodes.Items))
 }
 
-// initializeCNIPluginsPkgInfo populates the fields of cniPlugins
-func (f *wmcbFramework) initializeCniPluginsPkgInfo() error {
-	f.cniPlugins = pkgInfo{}
+//intitializePkgInfo populates the fields for cniPlugins package
+func (c *cniPlugins) initializePkgInfo() error {
+	var cniPluginsPkg cniPlugins
 
-	f.cniPlugins.url = cniPluginsBaseURL + f.TestFramework.LatestCniPluginsVersion + "/cni-plugins-windows-amd64-" +
-		f.TestFramework.LatestCniPluginsVersion + ".tgz"
+	cniPluginsPkg.url = cniPluginsBaseURL + framework.TestFramework.LatestCniPluginsVersion + "/cni-plugins-windows-amd64-" +
+		framework.TestFramework.LatestCniPluginsVersion + ".tgz"
 
-	cniPluginsSHA, err := framework.getLatestCniPluginsSHA()
+	cniPluginsSHA, err := cniPluginsPkg.getSHAValue(cniPluginsPkg.url)
 	if err != nil {
 		return fmt.Errorf("unable to get SHA for CNI Plugins: %v", err)
 	}
-	f.cniPlugins.sha = cniPluginsSHA
+	cniPluginsPkg.sha = cniPluginsSHA
 
-	f.cniPlugins.shaType = "sha512"
+	cniPluginsPkg.shaType = "sha512"
+	c.pkgInfo = cniPluginsPkg.pkgInfo
+	c.pkgName = cniPluginPkgName
```

Comment:
This does not look elegant. I suggest having a `newCNIPlugins` function that does all of this initialization. `initialize()` can be removed and should not be part of the interface. 

I also don't understand why `getSHAValue()` and `getSHAFileContent()` is part of the interface as I don't see it being used outside of `initialize()`.

I guess the question I am asking is, why is the interface needed? It is not like you are exporting it from external use.